### ISSUE TITLE
TestGrid outage due to bad config.proto upload

### ISSUE THREAD

**Issue by michelle192837** [kubernetes/test-infra#16625]
*tl;dr*: Grids on testgrid.k8s.io could not be viewed from ~11AM PST to ~4PM PST today, due to a bad config.proto upload that caused the frontend to panic on trying to load any grids.

*Details*: 
* Around 11:30 AM PST, some users noticed that trying to view tabs in testgrid.k8s.io was producing an "Internal Error" and no results.
* The frontend logs indicating that it was panicking on trying to determine the column_header type (see https://github.com/GoogleCloudPlatform/testgrid/blob/master/pb/config/config.proto#L96).
* After some investigation (the errors didn't seem to correspond to obvious update errors in the logs, or a TestGrid rollout), it seems that the specific issue was that blank column headers were uploaded in the config.proto (e.g. the resolved configuration from the .yaml files in configs/testgrids + annotations in ProwJobs), which are invalid for TestGrid.
* The issue was likely introduced by an update of TestGrid dependencies in test-infra (https://github.com/kubernetes/test-infra/pull/16597) that likely caused Configurator to upload a bad config.proto. The suspected PR was rolled back, and the configuration was regenerated with the rolled back code.
* Following that and a restart of external TestGrid around 4 PM, it seems that grids on testgrid.k8s.io are accessible (and, from random sampling, don't seem to have corrupted state).

Follow-up:
* PR #16624 to validate column_headers and prevent this happening again (thanks Erick!)
* Fix the underlying bad column_header issue
* (possibly) Have the frontend 'best-effort' try to show the underlying data, even with issues. (It may not help in this particular case, but can go over the code to see if there's easy improvements here).

/area testgrid
/kind oncall-hotlist

**Comment by michelle192837** [comment]
Think this is mitigated in a few ways now. (Thanks again Erick!)
